clion/src/main/java/org/sonarlint/intellij/clion/AnalyzerConfiguration.java

```diff
@@ -175,19 +175,33 @@ static ForcedLanguage getSonarLanguage(OCLanguageKind languageKind) {
   }
 
   private boolean usingRemoteOrWslToolchain(OCResolveConfiguration configuration) {
-    final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
-    CPPEnvironment cppEnvironment = null;
-    for (var initializedWorkspace : initializedWorkspaces) {
-      if (initializedWorkspace instanceof CMakeWorkspace cMakeWorkspace) {
-        cppEnvironment = getCMakeCppEnvironment(cMakeWorkspace, configuration);
-      } else {
-        cppEnvironment = tryReflection(initializedWorkspace);
-        if (cppEnvironment != null) {
-          break;
+    try {
+      tryIfClassAccessible();
+      final var initializedWorkspaces = CidrWorkspace.getInitializedWorkspaces(project);
+      for (var initializedWorkspace : initializedWorkspaces) {
+        if (Class.forName("com.jetbrains.cidr.cpp.cmake.workspace.CMakeWorkspace").isInstance(initializedWorkspace)) {
+          var cppEnvironment = getCMakeCppEnvironment(initializedWorkspace, configuration);
+          if (cppEnvironment != null) {
+            return cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker();
+          }
+        } else {
+          var cppEnvironment = tryReflection(initializedWorkspace);
+          if (cppEnvironment != null) {
+            return cppEnvironment.getToolSet().isRemote() || cppEnvironment.getToolSet().isWSL() || cppEnvironment.getToolSet().isDocker();
+          }
         }
       }
+      SonarLintConsole.get(project).debug("Not using remote or WSL toolchain");
+      return false;
+    } catch (ClassNotFoundException e) {
+      SonarLintConsole.get(project).debug("Could not check if remote or WSL toolchain is used");
+      return false;
```

Comment:
I prefer what you did in other places with a `isSupported` call that lets you return early from the function if needed, instead of a big try-catch